pex/pex_rules.bzl

```diff
@@ -149,6 +149,12 @@ def _gen_manifest(py, runfiles):
     dpath = f.short_path
     if dpath.startswith("../"):
       dpath = dpath[3:]
+
+    for prefix in strip_pex_path_prefixes:
+      if dpath.startswith(prefix):
```

Comment:
I think you have the same bug here that I had: a path can match two prefixes.

Consider strip_prefixes set to: `['aaa', 'bbb']`

If the files are the following:

* `aaa/bbb/foo` (expect to strip aaa and produce `bbb/foo`)
* `bbb/aaa/bar` (expect to strip bbb and produce `aaa/bar`)

However, depending on the order the prefixes are applied, you will probably get one of `foo` or `bar` as an output, since it strips both.

To fix it, in my version I strip the LONGEST prefix match, which is a bit annoying but makes in unambiguous, and works when there are "overlapping" strip prefixes (e.g. strip `root1` and `root1/foo/bar/subroot`)

Reply:
I am unsure if it's less unambiguous then stripping by the first matched prefix which is what this patch does.

Reply:
Ah crap I missed the `break` on the next line. Silly me, sorry.

Which one will be first though? It will depend on the order `deps` are specified in BUILD files. I guess that could be useful.

Reply:
Not `deps` but first in the `strip_pex_path_prefixes` list, the stripping afterall apply to all files in `runfiles` which can be either `deps`, `srcs` or even `data` attribute, if I recall well.

Reply:
Ah! I see: This adds the attribute to `pex_binary`. I see! I added it to `pex_library` since we wanted to have some `pex_library` make the python package `whatever`, no matter where it lives in the Bazel workspace.

Anyway: Great minds think alike! I agree that something like this should end up as part of these rules.

Reply:
What would be the benefit of adding this to `pex_library` versus `pex_binary`?

Reply:
It might be useful to have it apply to `data` from pex_library rules; otherwise you would have to manually propagate that information along to the final pex_binary in some situations.

Reply:
Hmm... Now that you mention it: I'm pretty sure my version of this does not actually consider the `data` attribute, and only applies it to the `srcs` attribute. I think the internal code where I'm using this doesn't use any data files.

Reply:
So I've been playing with this and I definitely think that applying it to the `pex_library` (both `srcs` and `data`) makes the most sense as it removes the need for duplicating code. In my case I have two binaries that make use of the same library and I'd have to strip the same prefix in both binaries. In Buck this is called `base_module` maybe we can adopt the same name (see: https://buckbuild.com/rule/python_library.html#base_module).

Reply:
A counter-argument will be that you wouldn’t include test data in the
`pex_library`? B/c if I understood correctly having this attribute in
`pex_library` only requires that you will need to define anything that
eligible to striping in the `pex_library` rule, correct?

On the other hand, having it in `pex_binary` and `pex_test` have direct effect
on the resulting Pex since this later are the rules that create them, and
yes, I agree that interface wise it's not DRY (and personally I don't like it too :)) however IMHO it gives users more control.